src/Pate/Monad.hs

```diff
@@ -423,8 +429,27 @@ withAssumptionSet asm f = withSym $ \sym -> withSolverProcess $ \sp -> do
   CMR.local (\env -> env { envCurrentFrame = Some (asm <> curAsm) }) $ do
     IO.withRunInIO $ \inIO -> WPO.inNewFrame sp $ do
       W4.assume (WPO.solverConn sp) p
+      inIO $ validateAssumptions curAsm asm
       inIO f
 
+validateAssumptions ::
```

Comment:
Is it the case that this is validating that the assumptions that are active in the solver are consistent (i.e., not inherently contradictory)? If so, I think it is worth noting that the `AssumptionSet`s passed in are *not* what is actually getting checked -- rather, they seem to be for error messages.

Reply:
Added a comment